File: app/src/main/java/com/inhealion/generator/GeneratorApp.java
package com.inhealion.generator;

import android.app.Application;

import com.inhealion.networking.GeneratorApiClient;
import com.inhealion.networking.account.SharedPrefAccountStore;

import timber.log.Timber;

public class GeneratorApp extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }

        GeneratorApiClient.initialize(BuildConfig.BASE_URL, new SharedPrefAccountStore(this));
    }
}

File: app/src/main/java/com/inhealion/generator/GeneratorApp.kt
package com.inhealion.generator

import android.app.Application
import com.inhealion.generator.data.di.repositoryModule
import com.inhealion.generator.di.*
import com.inhealion.generator.extension.initFirebase
import com.inhealion.generator.extension.initLogger
import com.inhealion.generator.extension.initRepository
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin
import org.koin.core.logger.Level

class GeneratorApp : Application() {
    override fun onCreate() {
        super.onCreate()
        startKoin {
            androidLogger(if (BuildConfig.DEBUG) Level.ERROR else Level.NONE)
            androidContext(this@GeneratorApp)
            modules(
                networkModule,
                eventModule,
                appModule,
                serviceModule,
                viewModelModule,
                repositoryModule,
                utilsModule
            )
        }

        initFirebase()
        initLogger()
        initRepository()
    }
}
